controllers/tokenupload_controller.go

```diff
@@ -61,67 +67,34 @@ type TokenUploadReconciler struct {
 func (r *TokenUploadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 
 	lg := log.FromContext(ctx)
-
 	uploadSecretList := &corev1.SecretList{}
 
 	// uploadSecretList secrets labeled with "spi.appstudio.redhat.com/upload-secret"
 	err := r.List(ctx, uploadSecretList, client.HasLabels{tokenSecretLabel})
 	if err != nil {
 		lg.Error(err, "can not get uploadSecretList of secrets")
-		return ctrl.Result{}, fmt.Errorf("can not get uploadSecretList of secrets: %w,", err)
+		return ctrl.Result{}, fmt.Errorf("can not get uploadSecretList of secrets: %w", err)
 	}
 
 	for i, uploadSecret := range uploadSecretList.Items {
 
-		var accessToken spi.SPIAccessToken
-
-		// we immediatelly delete the Secret
+		// we immediately delete the Secret
 		err := r.Delete(ctx, &uploadSecretList.Items[i])
 		if err != nil {
 			logError(ctx, uploadSecret, fmt.Errorf("can not delete the Secret: %w", err), r, lg)
 			continue
 		}
 
-		// if spiTokenName field is not empty - try to find SPIAccessToken by it
-		if uploadSecret.Labels[spiTokenNameLabel] != "" {
-			spiTokenName := uploadSecret.Labels[spiTokenNameLabel]
-			accessToken = spi.SPIAccessToken{}
-			err = r.Get(ctx, types.NamespacedName{Name: spiTokenName, Namespace: uploadSecret.Namespace}, &accessToken)
-
+		// try to find  SPIAccessToken
+		accessToken, err := findSpiAccessToken(ctx, uploadSecret, r, lg)
+		if err != nil {
+			continue
```

Comment:
We usually accompany the error handling with logging the error. Here, it looks like we're not doing that when in fact we are because the `findSpiAccessToken` does that. Maybe worth a comment here that it has already been logged and why it is ok to not react in any other way to it...

Reply:
I see, if it is an agreement, I can move logging here